src/main/java/com/robosh/ejournal/service/AdminService.java

```diff
@@ -1,8 +1,33 @@
 package com.robosh.ejournal.service;
 
+import com.robosh.ejournal.data.dto.admin.AdminInfoDto;
+import com.robosh.ejournal.data.dto.admin.UpdateAdminDto;
+import com.robosh.ejournal.data.entity.admin.Admin;
+import com.robosh.ejournal.data.mapping.admin.AdminInfoMapper;
+import com.robosh.ejournal.data.mapping.admin.UpdateAdminMapper;
+import com.robosh.ejournal.data.repository.AdminRepository;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import javax.validation.ValidationException;
+
 @Service
+@RequiredArgsConstructor
 public class AdminService {
 
+    private final AdminRepository adminRepository;
+    private final UpdateAdminMapper updateAdminMapper;
+    private final AdminInfoMapper adminInfoMapper;
+
+    public AdminInfoDto save(UpdateAdminDto updateAdminDto) {
+        if (!updateAdminDto.getPassword().equals(updateAdminDto.getConfirmedPassword())) {
+            throw new ValidationException("Password should be same");
+        }
+        Admin admin = updateAdminMapper.dtoToAdmin(updateAdminDto);
+        if (updateAdminDto.getId() == null) {
```

Comment:
maybe you should use Optional ?

Reply:
it is not optional problem. When id is null we should set school to null as super admin don't have school. If we don't do it, hibernate will try to find a school with id=null in the database and we will have an exception